File: src/NetMusicStore/Catalog.Domain/Album/Events/AlbumCreatedDomainEvent.cs
using Shared.Domain.Bus.Event;

namespace Catalog.Domain.Events
{
    public class AlbumCreatedDomainEvent : DomainEvent
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public int AlbumTypeId { get; set; }
        public string Description { get; set; }
        public string Author { get; set; }
        public string[] Tags { get; set; }

        public AlbumCreatedDomainEvent(int albumTypeId, string id, string name, string description, string author, string[] tags)
        {
            AlbumTypeId = albumTypeId;
            Id = id;
            Name = name;
            Description = description;
            Author = author;
            Tags = tags;
        }

        public override string EventName()
        {
            throw new NotImplementedException();
        }

        public override Dictionary<string, string> ToPrimitives()
        {
            throw new NotImplementedException();
        }

        public override DomainEvent FromPrimitives(string aggregateId, Dictionary<string, string> body, string eventId, string occurredOn)
        {
            throw new NotImplementedException();
        }
    }
}

File: src/BackOffice/server/src/services/albums/domain/AlbumAggregate/events/AlbumCreatedDomainEvent.ts
import { DomainEvent } from '../../../../shared/domain';

type CreateAlbumDomainEventBody = {
  readonly id: string;
  readonly eventName: string;
  readonly name: string;
  readonly description: string;
  readonly registerDate: Date;
  readonly author: string;
};

class AlbumCreatedDomainEvent extends DomainEvent {
  static readonly EVENT_NAME = 'album.created';

  readonly name: string;
  readonly description: string;
  readonly registerDate: Date;
  readonly author: string;

  constructor({
    id,
    name,
    description,
    registerDate,
    author,
    eventId,
    occurredOn
  }: {
    id: string;
    eventId?: string;
    name: string;
    description: string;
    registerDate: Date;
    author: string;
    occurredOn?: Date;
  }) {
    super(AlbumCreatedDomainEvent.EVENT_NAME, id, eventId, occurredOn);

    this.name = name;
    this.description = description;
    this.registerDate = registerDate;
    this.author = author;
  }

  toPrimitive(): CreateAlbumDomainEventBody {
    const { name, description, registerDate, author, aggregateId } = this;
    return {
      name,
      description,
      registerDate,
      author,
      eventName: AlbumCreatedDomainEvent.EVENT_NAME,
      id: aggregateId
    };
  }

  static fromPrimitives(
    aggregateId: string,
    body: CreateAlbumDomainEventBody,
    eventId: string,
    occurredOn: Date
  ): DomainEvent {
    return new AlbumCreatedDomainEvent({
      id: aggregateId,
      name: body.name,
      description: body.description,
      registerDate: body.registerDate,
      author: body.author,
      eventId,
      occurredOn
    });
  }
}

export default AlbumCreatedDomainEvent;
